src/api/polygons.ts

```diff
@@ -1,29 +1,47 @@
 import { TargomoClient } from './targomoClient'
 import { LatLngId } from '../index';
 import { PolygonRequestOptions } from '../types/options/polygonRequestOptions';
-import { PolygonRequestPayload } from './payload/polygonRequestPayload';
+import { PolygonRequestPayload, PolygonGeoJsonOptions, PolygonSvgOptions } from './payload/polygonRequestPayload';
 import { UrlUtil } from '../util/urlUtil';
 import { requests} from '../util/requestUtil';
+import { PolygonSvgResult } from '../types/responses/polygonSvgResult';
+import { FeatureCollection, MultiPolygon } from 'geojson';
+
+
 
 export class PolygonsClient {
   constructor(private client: TargomoClient) {
   }
+  /**
+   * Request geojson polygons for one or more sources from r360 service
+   * @param sources
+   * @param options
+   */
+  async fetch(sources: LatLngId[], options: PolygonGeoJsonOptions): Promise<FeatureCollection<MultiPolygon>>;
 
-  // TODO: maybe  have separate methods for svg and geojson so that return type can be different
 
   /**
-   * Request polygons for one or more sources from r360 service
+   * Request svg polygons for one or more sources from r360 service
+   * @param sources
+   * @param options
    */
-  async fetch(sources: LatLngId[], options: PolygonRequestOptions): Promise<{}> {
-    const url = UrlUtil.buildTargomoUrl(this.client.serviceUrl, 'polygon', this.client.serviceKey)
-    const cfg = new PolygonRequestPayload(this.client, sources, options)
-    const result = await requests(this.client, options).fetchCachedData(options.useClientCache, url, 'POST', cfg)
+  async fetch(sources: LatLngId[], options: PolygonSvgOptions): Promise<PolygonSvgResult>;
+
+  async fetch(sources: LatLngId[], options: PolygonSvgOptions|PolygonGeoJsonOptions):
+    Promise<PolygonSvgResult | FeatureCollection<MultiPolygon>> {
+      const cfg = new PolygonRequestPayload(this.client, sources, options)
+      const result = await this.executeFetch(sources, options, cfg);
+      if (options.serializer === 'json') {
+        return result as PolygonSvgResult;
+      } else if (options.serializer === 'geojson') {
+        return result as FeatureCollection<MultiPolygon>;
+      }
+  }
 
-    if (options.serializer == 'geojson') {
-      result.metadata = options
-      return result
-    } else {
-      throw new Error('TODO: svg parsing')
-    }
+  private async executeFetch(sources: LatLngId[], options: PolygonRequestOptions, cfg: PolygonRequestPayload): Promise<{}> {
```

Comment:
prefix with _ -> `_executeFetch`

in non ts environments there are no private functions and the common practice is to prefix private funcs with an _